client/my-sites/theme/controller.jsx

```diff
@@ -45,26 +45,25 @@ export function fetchThemeDetailsData( context, next ) {
 	const themeSlug = context.params.slug;
 	const theme = themeDetailsCache.get( themeSlug );
 
-	if ( theme ) {
+	const HOUR_IN_MS = 3600000;
+	if ( theme && ( theme.timestamp + HOUR_IN_MS > Date.now() ) ) {
 		debug( 'found theme!', theme.id );
 		context.store.dispatch( receiveThemeDetails( theme ) );
-		next();
+		return next();
 	}
 
 	themeSlug && wpcom.undocumented().themeDetails( themeSlug, ( error, data ) => {
```

Comment:
I think this is the last occurrence of a `wpcom.undocumented()` function call in our themes code that doesn't use promise syntax -- would be cool to convert it while we're here :-)